sled-agent/src/instance_manager.rs

```diff
@@ -574,20 +574,22 @@ impl InstanceManagerRunner {
         sled_identifiers: SledIdentifiers,
     ) -> Result<SledVmmState, Error> {
         let InstanceEnsureBody {
+            vmm_spec,
+            local_config,
             instance_id,
             migration_id,
             propolis_addr,
-            hardware,
             vmm_runtime,
             metadata,
         } = instance;
         info!(
             &self.log,
             "ensuring instance is registered";
+            "propolis_spec" => ?vmm_spec,
```

Comment:
This isn't quite right. The old `hardware` type wrapped some things (like the cloud-init bytes) in a `NoDebug` wrapper, but the generated `InstanceSpecV0` type has no such thing, so this is going to make for noisy logs (and there are some things we might just want to redact entirely, like Crucible VCR contents).

Since this is a newtype, we have the option of coming up with a just-descriptive-enough `Debug` impl for it (and can rely on Propolis logs to get more specific configuration data for a particular VMM instance).

Reply:
This is almost enough to make me wonder if it'd be better not to use propolis-client's generated types here. I'm going to experiment with that a bit.

Reply:
Using the native Propolis types does make this a lot cleaner. I've switched over to those in 9b44af3. This requires another Propolis client update, which is now oxidecomputer/propolis#899, and I'll need to fix up the Propolis commit hashes here once that merges.